stately-concurrent-collections/src/commonMain/kotlin/co/touchlab/stately/collections/ConcurrentMutableCollection.kt

```diff
@@ -0,0 +1,87 @@
+package co.touchlab.stately.collections
+
+import co.touchlab.stately.concurrency.Synchronizable
+import co.touchlab.stately.concurrency.synchronize
+
+class MyMutableData(private var count: Int = 0) : Synchronizable() {
+    fun add() {
+        synchronize { count++ }
+    }
+
+    val myCount: Int
+        get() = synchronize { count }
+}
+
+open class ConcurrentMutableCollection<E>(rootArg: Synchronizable? = null, private val del: MutableCollection<E>) :
+    Synchronizable(),
+    MutableCollection<E> {
+
+    internal val syncTarget: Synchronizable = rootArg ?: this
+
+    override val size: Int
+        get() = syncTarget.synchronize { del.size }
+
+    override fun contains(element: E): Boolean = syncTarget.synchronize { del.contains(element) }
+
+    override fun containsAll(elements: Collection<E>): Boolean = syncTarget.synchronize { del.containsAll(elements) }
+
+    override fun isEmpty(): Boolean = syncTarget.synchronize { del.isEmpty() }
+
+    override fun add(element: E): Boolean = syncTarget.synchronize { del.add(element) }
+
+    override fun addAll(elements: Collection<E>): Boolean = syncTarget.synchronize { del.addAll(elements) }
+
+    override fun clear() {
+        syncTarget.synchronize { del.clear() }
+    }
+
+    override fun iterator(): MutableIterator<E> =
+        syncTarget.synchronize { ConcurrentMutableIterator(syncTarget, del.iterator()) }
+
+    override fun remove(element: E): Boolean = syncTarget.synchronize { del.remove(element) }
+
+    override fun removeAll(elements: Collection<E>): Boolean = syncTarget.synchronize { del.removeAll(elements) }
+
+    override fun retainAll(elements: Collection<E>): Boolean = syncTarget.synchronize { del.retainAll(elements) }
+
+    fun <R> blockCollection(f: (MutableCollection<E>) -> R): R = syncTarget.synchronize {
+        val wrapper = MutableCollectionWrapper(del)
+        val result = f(wrapper)
+        wrapper.coll = mutableSetOf()
```

Comment:
Would it make sense to instead have the delegate in the wrapper nullable, set it to null here and crash when any of the wrappers are accessed when the delegate is null? This now feels like it's easier to misuse.